lms/static/js/instructor_dashboard/membership.js

```diff
@@ -153,28 +160,49 @@ such that the value can be defined later than this assignment (file load order).
                         class: 'revoke'
                     });
                     $revokeBtn.click(function() {
-                        return authlistwidgetreloadlist.modify_member_access(member.email, 'revoke', function(err) {
+                        authlistwidgetreloadlist.modify_member_access(member.email, 'revoke', function(err) {
                             if (err !== null) {
-                                return authlistwidgetreloadlist.show_errors(err);
+                                authlistwidgetreloadlist.show_errors(err);
                             }
                             authlistwidgetreloadlist.clear_errors();
-                            return authlistwidgetreloadlist.reload_list();
+                            authlistwidgetreloadlist.reload_list();
                         });
                     });
-                    return authlistwidgetreloadlist.add_row([member.username, member.email, $revokeBtn]);
+                    if (authlistwidgetreloadlist.rolename === 'Group Moderator') {
+                        if (member.group_name === null) {
+                            // There is No discussion division scheme selected so the Group Moderator role
+                            // should be disabled
+                            authlistwidgetreloadlist.list_enabled = false;
+                            $selectedOption = $('select#member-lists-selector').children('option:selected');
+                            if ($selectedOption.text() === 'Discussion Group Community TA') {
+                                authlistwidgetreloadlist.show_errors(
+                                    gettext('This role requires a divided discussions scheme.')
+                                );
+                                authlistwidgetreloadlist.$('input[type="button"].add').addClass('is-disabled');
+                            }
+                        } else {
+                            authlistwidgetreloadlist.list_enabled = true;
+                            authlistwidgetreloadlist.$('input[type="button"].add').removeClass('is-disabled');
```

Comment:
Nit: define authlistwidgetreloadlist.$('input[type="button"].add') as a variable at line 172 so that the selector doesn't have to be duplicated in the 2 conditionals.